nautobot/dcim/forms.py

```diff
@@ -459,7 +459,7 @@ def clean(self):
                         "would conflict with same-named devices in this rack."
                     }
                 )
-        return cleaned_data
+        return super().clean()
```

Comment:
In a lot of cases `form.clean()` doesn't actually return anything, as the "outcome" of the function is just to populate `self.cleaned_data`. What happens if we just remove this line?

Reply:
I think the actual issue is that we were never calling `super.clean()` in this function.

Reply:
Yep. The `CustomFieldModelFormMixin` clean method is what is actually taking the form custom field data and adding it to the instance. The `RackForm` clean method didn't exist. When it was added without `super().clean()` it introduced this bug.